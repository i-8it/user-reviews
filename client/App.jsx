import React, { Component } from "react";
import ReactDOM from "react-dom";
import ReviewEntry from "./Components/ReviewEntry/ReviewEntry.jsx";
import Search from "./Components/Search/Search.jsx";
import axios from "axios";
class App extends Component {
  constructor() {
    super();
    this.state = {
      allData: [],
      current: []
    };
    this.handleSearch = this.handleSearch.bind(this);
  }

  handleSearch(inputVal) {
    inputVal = inputVal.toLowerCase();
    var output = this.state.allData.filter(res => {
      return res.name.toLowerCase().includes(inputVal);
    });
    this.setState({
      current: output
    });
  }

  componentDidMount() {
    axios
      .get("/reviews")
      .then(response => {
        this.setState({
          allData: response.data
        });
      })
      .catch(error => {
        console.log(error);
      });
  }

  render() {
    console.log("current", this.state.current);
    return (
      <div>
        <Search handleSearch={this.handleSearch} />
        <ReviewEntry
          allData={this.state.allData}
          current={this.state.current}
        />
      </div>
    );
  }
}

ReactDOM.render(<App />, document.getElementById("root"));
