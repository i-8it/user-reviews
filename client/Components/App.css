@media screen and (max-width: 700px) {
  #wrapper {
    width: 60%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  #wrapper {
    width: 100% !important;
  }
}
