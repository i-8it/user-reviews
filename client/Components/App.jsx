/* global document */
import React, { Component } from "react";
import ReviewEntry from "./ReviewEntry/ReviewEntry.jsx";
import Search from "./Search/Search.jsx";
import axios from "axios";
class App extends Component {
  constructor() {
    super();
    this.state = {
      allData: [],
      current: []
    };
    this.handleSearch = this.handleSearch.bind(this);
  }

  handleSearch(inputVal) {
    inputVal = inputVal.toLowerCase();
    var output = this.state.allData.filter(res => {
      return res.name.toLowerCase().includes(inputVal);
    });
    this.setState({
      current: output
    });
  }

  componentDidMount() {
    axios
      .get("/reviews")
      .then(response => {
        this.setState({
          allData: response.data
        });
      })
      .catch(error => {
        console.log(error);
      });
  }

  render() {
    return (
      <div>
        <Search handleSearch={this.handleSearch} />
        <ReviewEntry
          allData={this.state.allData}
          current={this.state.current}
        />
      </div>
    );
  }
}

export default App;
