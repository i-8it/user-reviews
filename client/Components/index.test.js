import React from "react";
import { shallow } from "enzyme";
import App from "./App.jsx";

describe("<App />", () => {
  it("should render a <div>", () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });
});

// describe("<App />", () => {
//   it("renders without crashing", () => {
//     const wrapper = shallow(<App />).dive();
//   });
// });
