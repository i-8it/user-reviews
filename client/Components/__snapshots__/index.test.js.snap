// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render a <div> 1`] = `
<div>
  <h1>
    hello
  </h1>
  <Search
    handleSearch={[Function]}
  />
  <ReviewEntry
    allData={Array []}
    current={Array []}
  />
</div>
`;
